R/check_requires.R

```diff
@@ -6,44 +6,137 @@
 #'   [cli::cli_abort()]
 #'
 #' @importFrom rlang caller_env
 #' @noRd
-check_requires <- function(what, requires, where = c("CRAN", "Bioc")) {
+check_requires <- function(
+  what,
+  requires,
+  where = c("CRAN", "Bioc", "Python")
+) {
   where <- match.arg(where)
 
-  is_available <- map_lgl(requires, requireNamespace, quietly = TRUE)
+  if (where == "Python") {
+    check_python_packages(what, requires)
+  } else {
+    check_r_packages(what, requires, where)
+  }
+
+  invisible(TRUE)
+}
+
+#' Check required Python packages
+#'
+#' @param what A message stating what the packages are required for
+#' @param requires Character vector of required Python package names
+#'
+#' @importFrom rlang caller_env
+#' @noRd
+check_python_packages <- function(what, requires) {
+  # First check if reticulate is available
+  if (!requireNamespace("reticulate", quietly = TRUE)) {
+    cli_abort(
+      c(
+        "{what} requires the {.pkg reticulate} package",
+        "i" = "Install it using {.code install.packages(\"reticulate\")}"
+      ),
+      call = caller_env()
+    )
+  }
+
+  is_available <- map_lgl(requires, reticulate::py_module_available)
 
   if (any(!is_available)) {
     missing <- requires[!is_available]
-
     # nolint start object_usage_linter
-    missing_str <- paste0("\"", paste(missing, collapse = "\", \""), "\"")
-    if (length(missing) > 1) {
-      missing_str <- paste0("c(", missing_str, ")")
-    }
-    fun <- switch(
-      where,
-      CRAN = "install.packages",
-      Bioc = "install.packages(\"BiocManager\"); BiocManager::install"
+    missing_str <- format_python_package_list(missing)
+    # nolint end object_usage_linter
+
+    cli_abort(
+      c(
+        "{what} requires the Python {.pkg {missing}} package{?s}",
+        "i" = paste(
+          "To continue, install {cli::qty(missing)}{?it/them} using",
+          "{.code reticulate::py_install({missing_str})}"
+        )
+      ),
+      call = caller_env()
     )
+  }
+}
+
+#' Check required R packages
+#'
+#' @param what A message stating what the packages are required for
+#' @param requires Character vector of required R package names
+#' @param where Where to install the packages from. Either "CRAN" or "Bioc"
+#'
+#' @importFrom rlang caller_env
+#' @noRd
+check_r_packages <- function(what, requires, where) {
+  is_available <- map_lgl(requires, requireNamespace, quietly = TRUE)
+
+  if (any(!is_available)) {
+    missing <- requires[!is_available]
+    # nolint start object_usage_linter
+    missing_str <- format_r_package_list(missing)
+    install_fun <- get_r_install_function(where)
     # nolint end object_usage_linter
 
     cli_abort(
       c(
         "{what} requires the {.pkg {missing}} package{?s}",
         "i" = paste(
           "To continue, install {cli::qty(missing)}{?it/them} using",
-          "{.code {fun}({missing_str})}"
+          "{.code {install_fun}({missing_str})}"
```

Comment:
This could maybe be runnable?

```suggestion
          "{.run {install_fun}({missing_str})}"
```